GitUI/CommandsDialogs/SettingsDialog/CheckSettingsLogic.cs

```diff
@@ -65,41 +65,43 @@ public bool SolveLinuxToolsDir(string? possibleNewPath = null)
                 gitpath = possibleNewPath.Trim();
             }
 
-            foreach (var toolsPath in new[] { @"bin\", @"usr\bin\" })
+            foreach (string toolsPath in new[] { @"bin\", @"usr\bin\" })
```

Comment:
If you rewrite this, this loop could be moved down to where toolspath is used, no need to do the first two checks twice.
(Not requesting changes - I would have preferred the previous version for 3.5 even if this is a nice refactor.)

Reply:
`toolsPath` is used for replacing in `gitpath` on the next line.
But there is another bug: The replacing of `gitpath` cannot work in the subsequent loops.